Homework7/FightLogic.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Homework7
+{
+    class FightLogic
+    {
+        public static Unit GenerateFighter()
+        {
+            int randomIndex = Randomizer.RandomInt(0, 3);
+            Unit unit;
+
+            switch (randomIndex)
+                {
+                case 0:
+                    unit = new Warrior();
+                        break;
+
+                case 1:
+                    unit = new Archer();
+                    break;
+
+                default:
+                    unit = new Wizard();
+                    break;
+            }
+
+            return unit;
+        }
+
+        public static Unit GenerateFighter(FighterType? fighterType)
+        {
+            Unit unit;
+
+            switch (fighterType)
+            {
+                case FighterType.Warrior:
+                    unit = new Warrior();
+                    break;
+
+                case FighterType.Archer:
+                    unit = new Archer();
+                    break;
+
+                case FighterType.Wizard:
+                    unit = new Wizard();
+                    break;
+
+                default:
+                    unit = GenerateFighter();
+                    break;
+            }
+
+            return unit;
+        }
+
+        public static Unit FightOneOnOne(FighterType? fighter1Type = null, FighterType? fighter2Type = null)
```

Comment:
Good method but to specific to be in FightLogic class. Please move it to Main as UI method

Reply:
move some actions in separate methods